src/main/java/com/minecolonies/coremod/colony/buildings/modules/AbstractCraftingBuildingModule.java

```diff
@@ -141,6 +140,14 @@ protected boolean isRecipeCompatibleWithCraftingModule(final IToken<?> token)
     {
         final IGenericRecipe recipe = GenericRecipe.of(token);
         if (recipe == null) return false;
+        for (final CustomRecipe rec : CustomRecipeManager.getInstance().getRecipes(building.getJobName()))
+        {
+            if (rec.getRecipeStorage().equals(IColonyManager.getInstance().getRecipeManager().getRecipes().get(token)))
+            {
+                return true;
+            }
+        }

```

Comment:
This code block needs to be skipped when `building == null`.

But it also seems inefficient, `isRecipeCompatibleWithCraftingModule` gets called a **lot** to establish JEI recipe compatibility.  It would be best to move this code to the caller of this method rather than having it here.  I presume it's only here for the legacy recipe persistence upgrade?

Reply:
Actually, I was confusing it with another method, the JEI integration doesn't call this particular one directly so it probably shouldn't encounter `building == null`.  But I still think this check probably shouldn't be here, as it could confuse the recipe teaching (you shouldn't be able to teach the custom recipes).

Reply:
true, I'll move it into the serialization verification instead, I'm calling this to not remove the pre-taught recipes.